AlfrescoApp/View Controllers/Sync View Controller/Data Source/RepositoryCollectionViewDataSource.m

```diff
@@ -182,10 +173,17 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cell
     [nodeCell updateCellInfoWithNode:node nodeStatus:nodeStatus];
     [nodeCell registerForNotifications];
     
-    [nodeCell updateStatusIconsIsFavoriteNode:NO isSyncNode:isSyncOn isTopLevelSyncNode:isTopLevelNode animate:NO];
-    [favoriteManager isNodeFavorite:node session:self.session completionBlock:^(BOOL isFavorite, NSError *error) {
-        [nodeCell updateStatusIconsIsFavoriteNode:isFavorite isSyncNode:isSyncOn isTopLevelSyncNode:isTopLevelNode animate:NO];
-    }];
+    if ([self isKindOfClass:[FavoritesCollectionViewDataSource class]])
```

Comment:
A better way would be to extract the behaviour described by the else branch into a method in the parent class and then override it in the child class with the behaviour from the if branch.